components/ModSubmissionsEditFormFacility.vue

```diff
@@ -341,7 +346,121 @@ function submitForm(e: Event) {
         return
     }
 
-    //submitthedatahere(googlemapsURL, mapLatitude, mapLongitude)
-    console.log('Form submitted!!! woohoo')
+    const updateSubmissionId = formSubmissionId || ''
+
+    if (!updateSubmissionId) {
+        console.error('Facility ID is required for updating the facility')
+        return
+    }
+
+    const facilityInputVariables = {
+        updateSubmissionId,
+        input: {
+            isUnderReview: true,
+            facility: {
+                nameEn: nameEn.value || '',
+                nameJa: nameJp.value || '',
+                contact: {
+                    googleMapsUrl: googlemapsURL.value || '',
+                    email: email.value || '',
+                    phone: phone.value || '',
+                    website: website.value || '',
+                    address: {
+                        postalCode: postalCode.value || '',
+                        prefectureEn: prefectureEn.value || '',
+                        cityEn: cityEn.value || '',
+                        addressLine1En: addressLine1En.value || '',
+                        addressLine2En: addressLine2En.value || '',
+                        prefectureJa: prefectureJp.value || '',
+                        cityJa: cityJp.value || '',
+                        addressLine1Ja: addressLine1Jp.value || '',
+                        addressLine2Ja: addressLine2Jp.value || ''
+                    }
+                },
+                healthcareProfessionalIds: [],
+                mapLatitude: parseFloat(mapLatitude.value) || 0,
+                mapLongitude: parseFloat(mapLongitude.value) || 0
+            }
+        }
+    }
+
+    const maxAttempts = 3
+    let mutationAttempts = 0
+    let mutationSuccess = false
+
+    function delay(milliseconds: number) {
+        return new Promise(resolve => setTimeout(resolve, milliseconds))
+    }
+
```

Comment:
this loop handles the retries and state management based on where the file is saved.